src/Server/PostgreSQLHandler.cpp

```diff
@@ -416,6 +444,152 @@ void PostgreSQLHandler::processQuery()
+    return true;
+}
+
+bool PostgreSQLHandler::processExecute(const String & query, ContextMutablePtr query_context)
+{
+    auto parser = ParserExecute();
+    ASTPtr prepare;
+    try
+    {
+        prepare = parseQuery(parser, query, 0, DBMS_DEFAULT_MAX_PARSER_DEPTH, DBMS_DEFAULT_MAX_PARSER_BACKTRACKS);
+    }
+    catch (const Exception &)
+    {
+        return false;
+    }
+
+    auto result_query = prepared_statements_manager.getStatement(prepare->as<ASTExecute>());
+
+    CurrentThread::QueryScope query_scope{query_context};
+    ReadBufferFromString read_buf(result_query);
+    executeQuery(read_buf, *out, false, query_context, {});
+
+    PostgreSQLProtocol::Messaging::CommandComplete::Command command =
+        PostgreSQLProtocol::Messaging::CommandComplete::classifyQuery(result_query);
+    message_transport->send(PostgreSQLProtocol::Messaging::CommandComplete(command, 0), true);
+
+    return true;
+}
+
+bool PostgreSQLHandler::processDeallocate(const String & query)
+{
+    auto parser = ParserDeallocate();
+    ASTPtr deallocate;
+    try
+    {
+        deallocate = parseQuery(parser, query, 0, DBMS_DEFAULT_MAX_PARSER_DEPTH, DBMS_DEFAULT_MAX_PARSER_BACKTRACKS);
+    }
+    catch (const Exception &)
+    {
+        return false;
+    }
+
+    prepared_statements_manager.deleteStatement(deallocate->as<ASTDeallocate>());
+
+    PostgreSQLProtocol::Messaging::CommandComplete::Command command =
+        PostgreSQLProtocol::Messaging::CommandComplete::classifyQuery(query);
+    message_transport->send(PostgreSQLProtocol::Messaging::CommandComplete(command, 0), true);
+    return true;
+}
+
+void PostgreSQLHandler::processParseQuery()
+{
+    try
+    {
+        std::unique_ptr<PostgreSQLProtocol::Messaging::ParseQuery> query =
+            message_transport->receive<PostgreSQLProtocol::Messaging::ParseQuery>();
+
+        auto statement = std::make_shared<ASTPreparedStatement>();
+        statement->function_name = query->function_name;
+        statement->function_body = query->sql_query;
+        prepared_statements_manager.addStatement(statement.get());
+        message_transport->send(PostgreSQLProtocol::Messaging::ParseQueryComplete(), true);
+    }
+    catch (const Exception & e)
+    {
+        message_transport->send(
+            PostgreSQLProtocol::Messaging::ErrorOrNoticeResponse(
+                PostgreSQLProtocol::Messaging::ErrorOrNoticeResponse::ERROR, "2F000", "Query execution failed.\n" + e.displayText()),
+            true);
+        throw;
+    }
+}
+
+void PostgreSQLHandler::processBindQuery()
+{
+    try
+    {
+        std::unique_ptr<PostgreSQLProtocol::Messaging::BindQuery> query =
+            message_transport->receive<PostgreSQLProtocol::Messaging::BindQuery>();
+
+        auto statement = std::make_shared<ASTPreparedStatement>();
+        statement->function_name = query->function_name;
+        prepared_statements_manager.addStatement(statement.get());
```

Comment:
it seems incorrect - `prepared_statements_manager` will override mapping of function name to query with empty string - previously it was already populated in `PostgreSQLHandler::processParseQuery` by query. I believe it should be:
```suggestion
        prepared_statements_manager.attachBindQuery(query);
```